tests/drop_counters/configurable_drop_counters.py

```diff
@@ -0,0 +1,163 @@
+
+"""
+Utilities for interacting with configurable drop counters.
+"""
+
+from collections import namedtuple
+
+PORT_INGRESS_COUNTER_TYPE = "PORT_INGRESS_DROPS"
+SWITCH_INGRESS_COUNTER_TYPE = "SWITCH_INGRESS_DROPS"
+
+SUPPORTED_COUNTER_TYPES = [PORT_INGRESS_COUNTER_TYPE, SWITCH_INGRESS_COUNTER_TYPE]
+
+_ParserParameters = namedtuple("_ParserParameters", ["leading_rows",
+                                                     "header_rows",
+                                                     "status_columns"])
+_PARSER_PARAMETERS = {
+    PORT_INGRESS_COUNTER_TYPE: _ParserParameters(leading_rows=0, header_rows=2, status_columns=2),
+    SWITCH_INGRESS_COUNTER_TYPE: _ParserParameters(leading_rows=1, header_rows=2, status_columns=1)
+}
+
+_SHOW_CAPABILITIES = "show dropcounters capabilities"
+_CREATE_COUNTER = "config dropcounters install {} {} {}"
+_DELETE_COUNTER = "config dropcounters delete {}"
+
+def get_device_capabilities(dut):
+    """
+    Fetches the drop counter capabilities for the given device.
+
+    Args:
+        dut (SonicHost): The device to query for drop counter capabilities.
+
+    Returns:
+        A Dictionary containing 1) A Dictionary that maps from supported counter types to the
+        number of support counters of that type, and 2) a Dictionary that maps from supported
+        counter types to drop reasons that are supported for that type.
+
+        If drop counters are not supported on this device, then an empty dictionary will be
+        returned, like this: `{"counters": {}, "reasons": {}}`.
+
+        If any other error occurs, then None will be returned.
+    """
+
+    output = dut.command(_SHOW_CAPABILITIES)
+
+    if output["rc"] == 2:
+        return {"counters": {}, "reasons": {}}
+    elif output["rc"]:
+        return None
+
+    if "does not support drop counters" in output["stdout"]:
+        return {"counters": {}, "reasons": {}}
+
+    counters = {}
+    reasons = {}
+
+    # Delete the header.
+    output = output["stdout_lines"][2:]
+
+    # Extract the supported counter types and quantities from the first
+    # part of the output. We'll parse until we hit the empty line.
+    line = output.pop(0)
+    while line:
```

Comment:
You can use while len(output): to loop, then you can move the output.pop(0) into the while loop. Save one duplicate line.

Reply:
I see, you are stopping at empty line. never mind.